modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBDecision.java

```diff
@@ -52,8 +73,15 @@ public Expression decisionNot(){
      *          ["!=", null, null]: boolean
      * @return
      */
-    public Expression decisionNotEqual(){
-        return null;
+    private Expression decisionNotEqual() {
+        // validate the arg list
+        if (json.size() != 3) {
+            throwUnexpectedArgumentCount(NOT_EQUALS, 2);
+        }
+        // get the comparables
+        Expression comparable1 = parse.string(json, 1);
+        Expression comparable2 = parse.string(json, 2);
+        return ff.function("mbNotEqualTo", comparable1, comparable2);
```

Comment:
Have you tried FilterFunction_notEqualTo / "notEqualTo" function, or will it not work with this for a certain reason?

Reply:
Yes. That function works except when either of the arguments is `null`. It has a short circuit that returns `false` if either is `null`. The contract for MB's Decisions is that `null`s are equal, so the function I created treats 2 `null`s as equal. Also, the existing functions compare Objects using their `toString()` method, when the class types are not equal, which causes Boolean values to be equal to the associative string values (i.e. Boolean.TRUE ends up being equal to "true"). According to the MBExpression contract, two values are not equal if their types aren't equal.

Reply:
Sounds good.

Reply:
For the null issue, I wonder if that is unintended functionality in the GeoTools function - it may be worthwhile to raise a community bug report around that. Your solution here seems good for MBStyle though